rcl/src/rcl/logging.c

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl/logging.h"
+
+bool g_rcl_logging_initialized = false;
+
+RclLogFunction g_rcl_logging_output_handler = NULL;
+
+int g_rcl_logging_severity_threshold = 0;
+
+void rcl_logging_initialize()
+{
+  if (!g_rcl_logging_initialized) {
+    g_rcl_logging_initialized = true;
```

Comment:
I was thinking about this last night, and I think if that you might ought to move this line (assigning true to the global bool) to the end of the if statement block.

The reason I think that is that I think this bool is there to protect logging functions so they don't try to use logging output handler and severity threshold before they are assigned. So, if a logging macro is executed between lines 32 and 33, then they might try to use them anyways even though they are not set yet in this thread.